### PR TITLE
Change modify_disabled to create_disabled and delete_disabled

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-1226

### CODE REVIEW

File: pkg/api/oauthrelyingparty/provider.go [hunk 1]
```diff
@@ -32,15 +32,26 @@ func (c ProviderConfig) ClientID() string {
...
+		c["delete_disabled"] = defaultValue
+	}
+}
+
+func (c ProviderConfig) CreateDisabled() bool {
+	modify_disabled, _ := c["create_disabled"].(bool)
+	return modify_disabled
+}
+func (c ProviderConfig) DeleteDisabled() bool {
+	modify_disabled, _ := c["delete_disabled"].(bool)
```
Comment (louischan-oursky) [inline pkg/api/oauthrelyingparty/provider.go]: ModifyDisabled is removed. But for the following case, it should be read.

The actual config
```yaml
modify_disabled: true
```
This project has configured modify_disabled=true for a particular provider.

Comment (tung2744) [reply pkg/api/oauthrelyingparty/provider.go]: @louischan-oursky I think this part should be handled by the implementation of the provider.

Something like:

```go
func (ProviderABC) SetDefaults(cfg oauthrelyingparty.ProviderConfig) {
	deprecatedModifyDisabled, ok := cfg["modify_disabled"].(bool)
	if ok {
		cfg.SetDefaultsUpdateDisabled(deprecatedModifyDisabled)
	}
}
```

Comment (louischan-oursky) [reply pkg/api/oauthrelyingparty/provider.go]: According to our offline discussion, we will remove `Alias()`, `SetDefaultsModifyDisabledFalse()`, and `ModifyDisabled()`

Comment (tung2744) [reply pkg/api/oauthrelyingparty/provider.go]: I pushed a new commit for this change, but I am still working on the authgear-server changes.

**Pull request merged**